interface/super/manage_site_files.php

```diff
@@ -99,6 +100,42 @@
   }
 
 }
+
+/**
+ * Thumbnails generator
+ * generating  thumbnail image to all images files from documents table
+ */
+
+switch((int)$GLOBALS['document_storage_method']){
+    case 0:
+        $storage_method = 'on Disc';
+        break;
+    case 1:
+        $storage_method = 'in CouchDB';
+        break;
+}
+
+if(isset($_POST['generate_thumbnails'])) {
+
+    $thumb_generator = new ThumbnailGenerator();
+    $results = $thumb_generator->generate_all();
+
+    $thumbnail_msg = "<p style='color: green'>" . xl('Generated thumbnail(s)') . " : {$results['sum_success']}</p>";
+    $thumbnail_msg .= "<p style='color: red'>" . xl('Failed to generate') . ": {$results['sum_failed']}</p>";
+    foreach($results['failed'] as $key => $file){
+        $num = $key +1;
+        $thumbnail_msg .= "<p style='color: red; font-size: 11px'>$num.". " .$file</p>";
```

Comment:
for new code, need to escape correctly:
``````
$thumbnail_msg .= "<p style='color: red; font-size: 11px'>" . text($num) . ". ."  . text($file) . "</p>";